Content.Server/GameObjects/Components/Nutrition/CigSmokingComponent.cs

```diff
@@ -0,0 +1,96 @@
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Log;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Nutrition
+{
+    [RegisterComponent]
+    public class CigSmokingComponent : SharedSmokingComponent, IInteractUsing
+    {
+        private SmokingStates _currentState;
+        private ItemComponent _item;
+
+        [ViewVariables]
+        public SmokingStates CurrentState
+        {
+            get => _currentState;
+            private set
+            {
+                _currentState = value;
+                if (_item == null
+                    || !_item.Owner.TryGetComponent<SpriteComponent>(out SpriteComponent spriteComponent))
+                {
+                    Logger.Warning("No sprite component found for _item");
+                    return;
+                }
+                else
+                {
+                    _item.Owner.TryGetComponent(out ClothingComponent clothing);
+                    switch (_currentState)
+                    {
+                        case SmokingStates.Lit:
+                            _item.EquippedPrefix = "lit";
+                            clothing.ClothingEquippedPrefix = "lit";
+                            spriteComponent.LayerSetState(0, "lit-icon");
+                            break;
+                        default:
+                            _item.EquippedPrefix = "unlit";
+                            clothing.ClothingEquippedPrefix = "unlit";
+                            spriteComponent.LayerSetState(0, "icon");
+                            break;
+                    }
+
+                    if (Owner.TryGetComponent(out AppearanceComponent appearance))
+                    {
+                        appearance.SetData(SmokingVisuals.Smoking, _currentState);
+                    }
+                }
+            }
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            if (Owner.TryGetComponent<ItemComponent>(out _item))
+            {
+                _item.EquippedPrefix = "unlit";
+            }
+
+            _currentState = SmokingStates.Unlit;
+        }
+
+        public async Task<bool> InteractUsing(InteractUsingEventArgs eventArgs)
+        {
+            if (eventArgs.Using.TryGetComponent(out IHotItem lighter)
+                // TODO uncomment this is for ease of testing
+                //&& lighter.IsCurrentlyHot()
+                //&& cigarEntity._smokingStates == SmokingStates.Unlit
+            )
+            {
+                ToggleItem();
+                return true;
+            }
+
+            return false;
+        }
+
+        private void ToggleItem()
+        {
+            if (CurrentState == SmokingStates.Lit)
+            {
+                CurrentState = SmokingStates.Unlit;
+            }
+            else
+            {
+                CurrentState = SmokingStates.Lit;
+            }
+        }
```

Comment:
This is a very bad idea. Cigarettes aren't the kinds of items whose state you toggle... They have a clear "linear" path.
`Unlit -> Lit -> Butt`. Once it's lit, it can only turn to a cigarette butt.

Reply:
This is a WIP. I probably deleted the comment saying that this function is DEV only.

I left it like that since it's easier to debug toggling states than to test a cigar throw it away, spawn another, etc.

Reply:
E-cigarettes can be toggled

Reply:
@brndd I think a separate `WapeComponent` would be more applicable. For now, this should be enough imo.